### PR TITLE
OV-3: add sign up flow

### PR DESCRIPTION
**Pull request by JPjok3r**
Added migration file to create name field in users table
Modified user model and service to save user
Encryption service used
Endpoint /auth/sign-up added, returning object excluding passwords fields
validation checked with the FE code

### CODE REVIEW

File: backend/src/migrations/20240820093010_add_name_field_to_users_table.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+const TABLE_NAME = 'users';
+
+async function up(knex: Knex): Promise<void> {
+    await knex.schema.table(TABLE_NAME, (table) => {
+        table.string('name').notNullable();
+    });
+}
+
+function down(knex: Knex): Promise<void> {
+    return knex.schema.dropTableIfExists(TABLE_NAME);
```
Comment (anton-otroshchenko) [inline backend/src/migrations/20240820093010_add_name_field_to_users_table.ts:12]: Just drop the field not the table

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

File: backend/src/migrations/20240820093010_add_name_field_to_users_table.ts (hunk 1 above)
Comment (JPjok3r) [reply backend/src/migrations/20240820093010_add_name_field_to_users_table.ts:12]: tanks for the observation, I'll change it.

Comment (JPjok3r) [reply backend/src/migrations/20240820093010_add_name_field_to_users_table.ts]: thanks Anton, I fixed it.

**Review by stefano-lacorazza — DISMISSED** [review]

**Review by anton-otroshchenko — DISMISSED** [review]

**Review by sergiy4 — APPROVED** [review]

File: backend/src/bundles/users/user.entity.ts [hunk 2]
```diff
@@ -5,58 +5,69 @@ class UserEntity implements Entity {
 
     private 'email': string;
 
+    private 'name': string;
```
Comment (nikita-remeslov) [inline backend/src/bundles/users/user.entity.ts:8]: in design looks like we have Full Name

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: shared/src/bundles/users/validation-schemas/user-sign-up.validation-schema.ts [hunk 3]
```diff
@@ -3,23 +3,50 @@ import { z } from 'zod';
...
 type UserSignUpRequestValidationDto = {
+    name: z.ZodString;
     email: z.ZodString;
     password: z.ZodString;
+    confirmPassword: z.ZodString;
 };
 
 const userSignUp = z
     .object<UserSignUpRequestValidationDto>({
+        name: z.string().trim(),
```
Comment (nikita-remeslov) [inline shared/src/bundles/users/validation-schemas/user-sign-up.validation-schema.ts:14]: We should validate, that name have at least 2 words

**Comment by JPjok3r** [comment]
Hi @nikita-remeslov I checked and finished with the observation you made, when you have time you can check it, I based on the front-end code for:
shared/src/bundles/users/enums/user-validation-message.enum.ts
shared/src/bundles/users/enums/user-validation-rule.enum.ts

File: backend/src/bundles/auth/auth.controller.ts [hunk 4]
```diff
@@ -53,6 +53,8 @@ class AuthController extends BaseController {
      *                  format: email
      *                password:
      *                  type: string
+     *                name:
```
Comment (anton-otroshchenko) [inline backend/src/bundles/auth/auth.controller.ts:56]: also change it to full name here

Comment (JPjok3r) [reply backend/src/bundles/auth/auth.controller.ts]: Thanks Anton, it's done.

**Review by anton-otroshchenko — DISMISSED** [review]

**Comment by JPjok3r** [comment]
Hi @nikita-remeslov , @sofiia-trokhymchuk checked that `name` changed to `fullName`, checked that when trying to create a new user with an `email` that already exists show a better message, you can check it

**Review by anton-otroshchenko — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**